src/main/kotlin/com/seokjoo/todo/common/common/jwt/JwtAuthFilter.kt

```diff
@@ -0,0 +1,54 @@
+package com.seokjoo.todo.common.common.jwt
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.seokjoo.todo.common.exception.TodoException
+import com.seokjoo.todo.common.exception.TodoExceptionType
+import com.seokjoo.todo.presentation.error.ApiErrorResponse
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.web.filter.OncePerRequestFilter
+
+class JwtAuthFilter(
+    private val jwtProvider: JwtProvider,
+    private val objectMapper: ObjectMapper,
+) : OncePerRequestFilter() {
+    /**
+     * "/api/v1/auth/refresh" 는 controller 쪽에서 검증을 따로 하기 때문에 filter 해주지 않았습니다
+     */
+    override fun shouldNotFilter(request: HttpServletRequest): Boolean {
+        val excludePatterns =
+            listOf("/api/v1/auth/login", "/api/v1/auth/signup", "/api/v1/auth/refresh", "/api/v1/auth/delete")
+        return excludePatterns.any { request.requestURI.startsWith(it) }
+    }
+
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain,
+    ) {
+        try {
+            val token = request.getBearerToken()
+
+            val isNotValid = jwtProvider.validateToken(token).not()
+            if (isNotValid) throw throw TodoException.of(TodoExceptionType.AUTH_REFRESH_TOKEN_NOT_VALID)
```

Comment:
```suggestion
            if (isNotValid) throw TodoException.of(TodoExceptionType.AUTH_REFRESH_TOKEN_NOT_VALID)
```

요기 throw throw가 맞나요 ?!

Reply:
ㅋㅋ 너무 던져버리고 싶은 마음 ,, a70c6c51b6a113f0ebde3e9a312f8aa04d73dca3 수정했어요 !